packages/nilaway/package.yaml

```
@@ -0,0 +1,20 @@
+---
+name: nilaway
+description: |
+  NilAway is a static analysis tool that seeks to help developers avoid
+  nil panics in production by catching them at compile time rather than
+  at runtime.
+homepage: https://github.com/uber-go/nilaway
+licenses:
+  - Apache-2.0
+
+languages:
+  - Go
+categories:
+  - Linter
+
+source:
+  id: pkg:golang/go.uber.org/nilaway/cmd/nilaway@a267567c6ffff900df0c3394d031ee70079ec8df
```

According to the CI test suites this seems to already work but I believe Renovate will struggle with a commit SHA as the version specifier, so it'd be better to use the latest published version instead:
```suggestion
  id: pkg:golang/go.uber.org/nilaway@v0.0.0-20231117175943-a267567c6fff#cmd/nilaway
```

Also moved the subpath `cmd/nilaway` to the `#fragment` component of the PURL as `go.uber.org/nilaway` is the canonical package name